### ISSUE TITLE
[AG Grid] [Accessibility][JAWS] Filter option are not read aloud by screen reader unless it is selected first

### ISSUE THREAD

**Issue by jonaskuntoro** [ceolter/ag-grid#9163]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => see 'Providing a Reproducible Scenario'
[] feature request => do not use Github for feature requests, see 'Customers of AG Grid'
[] support request => see 'Requesting Community Support'
```

**Customers of AG Grid**

If you are a customer you are entitled to use the AG Grid's customer support system (powered by Zendesk). Please use that channel for guaranteed response from the AG Grid team with regards bugs, feature requests and support.

**Requesting Community Support**

If you are not a customer of AG Grid, ag-grid staff will label your issue as managed-by-the-community. This means that AG Grid staff is not going to be actively looking into it and it will get closed if inactive for more than one month. The community is welcome to help with this question/support issue.

**Providing a Reproducible Scenario**
Accepted reproducible scenarios are
- A description of the detailed steps to reproduce your behaviour in one of our examples in the docs.
- A plunker

If you decide to send us a plunker, from any example in our website use the plunker button in there to fork your own code by following the steps below:

- Select the framework that is appropriate to you from the drop-down
- Open it in plunker. (Use the button plunker in our example)
- Add your changes so that the behaviour is reproduced
- Save and Freeze the plunker(On the top left corner)
- Send us the link to the plunker(You can copy the URL from the browser)

If reporting a bug make sure to state.

Current behaviour.
Expected behaviour. If possible back this up with our docs/examples if possible

**Current behavior**
While we are navigating to the filter option using JAWS screen reader, the options are not announced by the screenreader. The behaviour is different when NVDA screen reader is used, it properly announces every single option. The option only will be announced  once an option is selected. Even when we try to re-navigate to the options, they are still not announced. (I have attached a video on how to reproduce it) **Note that : This also happens in demo version of AG Grid**

Steps:
1. Navigate using JAWS Screen reader
2. Go to filter option button on any column
3. Open the options
4. Navigate through all options
5. They are not announced
6. Select an option
7. That particular selected option is announced
8. Re-navigate through all options
9. They are still not announced

**Expected behavior**
The options should be announced regardless of the Screen readers or browsers. (I attached an img of how it looks when using NVDA on Edge)

**Please tell us about your environment:**

* **AG Grid version:** 31.3.1

* **Browser:**
Edge, Chrome, Firefox

* **Language:** Typescript

https://github.com/user-attachments/assets/73cbebe6-06a6-422e-8c43-88789e8de0f4

![image](https://github.com/user-attachments/assets/5095d659-3ee9-4350-98eb-31b6d66de887)

**Comment by AG-Zoheil** [comment]
Hi,

We've attempted to reproduce this using https://plnkr.co/edit/zOad4eyFBsPHVZQ2?open=main.js with NVDA in our latest version (33.0.3) and have been unable to do so. All options where announced correctly.

Please upgrade to our latest version and if an issues exists solely with a specific screen reader please contact their support team as it is most likely something they need to investigate.

Kind regards,
Zoheil

**Comment by jonaskuntoro** [comment]
Hi! The issue occurs only while using JAWS (it works just fine with NVDA, just like I mentioned in the description). In this case, I still wonder if the issue comes from the grid or the specific screen reader? @AG-Zoheil 

![image](https://github.com/user-attachments/assets/8be07dc9-da02-4382-b3c8-2601d413e2d6)

**Comment by AG-Zoheil** [comment]
Hello,

Thank you for getting back. We have added this to our backlog to investigate and we are tracking it with the following reference: **AG-13786**

We try to fix bugs from one release to the next, so this should be fixed in the next release or the one after if it was raised too close to the next release date.

You can follow the status of this item on the product pipeline page:
https://ag-grid.com/pipeline/

The best way to track this is to sign up for AG Grid new release notifications using [the instructions here](https://ag-grid.zendesk.com/hc/en-us/articles/360002648058-How-do-I-sign-up-for-ag-Grid-release-notifications-). This way you'll know as soon as a new version is out and you can check whether this specific item was implemented on [the changelog page](https://www.ag-grid.com/ag-grid-changelog/).

Thanks again for bringing this up with us.

Kind regards,
Zoheil
